core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -951,6 +951,11 @@ abstract class Record(private[chisel3] implicit val compileOptions: CompileOptio
             " See the cookbook entry 'How do I deal with the \"unable to clone\" error?' for more details."
         )
       }
+      if (field.binding.isDefined) {
+        throw RebindingException(
+          s"Cannot create Record ${this.className}; element ${field} of record is already a bound hardware."
```

Comment:
we can use terminology here and/or refer people to this link: https://www.chisel-lang.org/chisel3/docs/explanations/chisel-type-vs-scala-type.html

Reply:
and that is the sort of error messages you usually get:

```
// chisel3.package$ExpectedChiselTypeException: wire type '_44_Anon.hardware: Wire[MyBundle]' must be a Chisel type, not hardware
```